### ISSUE TITLE
Unexpected or hard to parse compile-time errors in switch-pattern statements

### ISSUE THREAD

**Issue by chloestefantsova** [dart-lang/sdk#50779]
The error was encountered during work on https://dart-review.googlesource.com/c/sdk/+/276524.

The following test results in a surprising error message.
* co19/LanguageFeatures/Patterns/logical_and_A02_t02

The text of the message is as follows:

```
tests/co19/src/LanguageFeatures/Patterns/logical_and_A02_t02.dart:27:5: Error: Switch case may fall through to the next case.
    case [> 0 && <= 2, 3]:
    ^
```

AFAIK, cases in pattern-switch statements can't fall through, and that was surprising to see.

This issue is assigned label `area-front-end` tentatively, and it needs further investigation, and possibly reassignment of the tags.

**Comment by johnniwinther** [comment]
The error message was report because the type analyzer wasn't notified that the patterns feature was enabled. This was fixed in https://dart-review.googlesource.com/c/sdk/+/278746
